### ISSUE TITLE
Poor handling of error responses during create operation

### ISSUE THREAD

**Issue by candlerb** [lxc/pylxd#77]
In `pylxd/container.py`:

~~~
    @classmethod
    def create(cls, client, config, wait=False):
        """Create a new container config."""
        response = client.api.containers.post(json=config)

        if wait:
            Operation.wait_for_operation(client, response.json()['operation'])
        return cls(name=config['name'])
~~~

However when there is an error response, it has no 'operation' member, as shown in the [API](https://github.com/lxc/lxd/blob/master/doc/rest-api.md#error)

As a result, the exception generated is not helpful in diagnosing the problem:

~~~
  File \"/usr/lib/python2.7/dist-packages/pylxd/container.py\", line 68, in create
    Operation.wait_for_operation(client, response.json()['operation'])
KeyError: 'operation'
~~~

An example of this is in [the new ansible lxd module](https://github.com/ansible/ansible-modules-extras/pull/2208#issuecomment-219454881)

**Comment by candlerb** [comment]
Very temporary workaround:

~~~
--- a/pylxd/container.py
+++ b/pylxd/container.py
@@ -63,6 +63,8 @@ class Container(mixin.Waitable, mixin.Marshallable):
     def create(cls, client, config, wait=False):
         """Create a new container config."""
         response = client.api.containers.post(json=config)
+        if response.status_code >= 400:
+            raise RuntimeError(repr(response.json()))

         if wait:
             Operation.wait_for_operation(client, response.json()['operation'])
~~~

**Comment by jpic** [comment]
Same error with `tox -e integration` here.

**Comment by jpic** [comment]
Agreed, this would remove boilerplate added by [573a53](https://github.com/lxc/pylxd/pull/80/commits/573a53d01ba8fa58dc258613047af0af00b99d53).

**Comment by rockstar** [comment]
Thanks for your report. I purposefully didn't catch most errors in the newer API initially, as the deprecated API over-catches errors. These are the kinds of reports that help us get a little more judicial in our error handling.  I'll get to work on this soon.

**Comment by rockstar** [comment]
Fixed in #86
